mkosi/__init__.py

```diff
@@ -4126,28 +4127,27 @@ def secure_boot_sign(
         return
 
     with mount():
-        for path, _, filenames in os.walk(root / "efi"):
-            for i in filenames:
-                if not i.endswith(".efi") and not i.endswith(".EFI"):
-                    continue
+        for f in itertools.chain(efi_path.glob('**/*.efi'), efi_path.glob('**/*.EFI')):
+            if os.path.exists(f"{f}.signed"):
+                MkosiPrinter.info(f"Not overwriting existing signed EFI binary {f}.signed")
+                continue
 
-                with complete_step(f"Signing EFI binary {i} in ESP…"):
-                    p = os.path.join(path, i)
-
-                    run(
-                        [
-                            "sbsign",
-                            "--key",
-                            args.secure_boot_key,
-                            "--cert",
-                            args.secure_boot_certificate,
-                            "--output",
-                            p + ".signed",
-                            p,
-                        ],
-                    )
+            with complete_step(f"Signing EFI binary {f}…"):
+                run(
+                    [
+                        "sbsign",
+                        "--key",
+                        args.secure_boot_key,
+                        "--cert",
+                        args.secure_boot_certificate,
+                        "--output",
+                        f"{f}.signed",
+                        f,
+                    ],
+                )
 
-                    os.rename(p + ".signed", p)
+                if replace:
+                    os.rename(f"{f}.signed", f)
```

Comment:
```suggestion
                    f.rename(f.with_suffix(".efi.signed"))
```

Reply:
Does this need to preserve the case (efi vs EFI)? See previous comment if it should.

Reply:
I don't think it does, but see above.